posts/2022-07-20-liberty-tools-eclipse.adoc

```diff
@@ -0,0 +1,94 @@
+
+== Language Server Integration
+
+Configuring your Liberty server is now simpler than ever with server config auto-complete, validation, and "hover over" descriptions.
+
+Additionally, you can quickly develop MicroProfile applications with automatic code blocks, auto-complete, and real time MP syntax validation. 
+
+
+== Try it out
+
+=== Installation
+
+Install the plugin using the instructions found link:https://github.com/OpenLiberty/liberty-tools-eclipse/blob/main/docs/installation/installation.md[here]
+
+=== Using the Dashboard
+
+Once installed, you will see the Open Liberty UFO logo in the top tray. Click on that to open the Liberty Dashboard tab:
+
+[.img_border_light]
+image::/img/blog/liberty-tools-dash-icon.png[Liberty Dashboard Icon,width=70%,align="center"]
+
+The Dashboard will automatically populate with any Maven and Gradle projects in your workspace that are configured with the Liberty Maven / Liberty Gradle plugins. If you do not have any apps in your current workspace, try importing one like the Open Liberty "Getting Started" app link:https://github.com/openliberty/guide-getting-started[here]. Once imported, refresh the Dashboard tab by clicking the refresh icon in the top right. 
+
+[.img_border_light]
+image::/img/blog/liberty-tools-dashboard.png[Liberty Dashboard,width=70%,align="center"]
+
+You should see your app listed in the dashboard. Right click on the app and you will see a dropdown with a list of actions you can perform.
+
+[.img_border_light]
+image::/img/blog/liberty-tools-actions.png[Actions Menu,width=70%,align="center"]
+
+Select the "Start" action to run your app in dev mode. 
+
+[.img_border_light]
+image::/img/blog/liberty-tools-terminal-start.png[Start in Dev Mode,width=70%,align="center"]
+
+Once running, right click on the app in the dashboard again and select "Run tests". 
+
+[.img_border_light]
+image::/img/blog/liberty-tools-test-action.png[Run Tests,width=70%,align="center"]
+
+Any automated tests associated with the app are run:
+
+[.img_border_light]
+image::/img/blog/liberty-tools-terminal-test.png[Tests Complete,width=70%,align="center"]
+
+=== Language Server Support
+
+Open `src/main/liberty/config/server.xml`. Start adding a new feature to the Feature Manager list. You will see a Content Assist pop up menu with auto-complete options:
+
+[.img_border_light]
+image::/img/blog/liberty-tools-feature-autocomplete.png[Feature Auto-complete,width=70%,align="center"]
+
+To get a list of available features and their descriptions, engage Content Assist by entering `Ctrl+Space` from within the feature config element:
+
+[.img_border_light]
+image::/img/blog/liberty-tools-feature-list.png[Feature List,width=70%,align="center"]
+
+You will see a complete list of Open Liberty features to choose from. 
```

Comment:
"a complete list of Open Liberty features"

Hmm.. maybe we should be a bit careful since OL vs WL features differs.   Maybe "a complete list of features depending on your Liberty install and version"?  

Is that true?  Does the feature description differ in OL vs WL?  Should we confirm?

Reply:
Good question.  I'll confirm that but for the sake of this blog I'm just going to remove that line altogether.